src/processors/engine.c

```diff
@@ -440,7 +440,7 @@ int Sagan_Engine ( _Sagan_Proc_Syslog *SaganProcSyslog_LOCAL, bool dynamic_rule_
 
                     match = false;
```

Comment:
match is set here.  At the "program",  the test will always be true.  

We want these to act like an "and" statement.  For example,  if "program" and "facility" match, then fire.

Reply:
Yes indeed: I put it there to make it consistent with all the following tests, so that they can be moved around.  But I can remove the first one if you like.  The subsequent ones must remain.

The reason for this change is exactly to make it work as "and", which it doesn't today.

Reply:
Yep.  Just tested and you are 100% correct.   Thanks for catching that.    I was able to reproduce and the patch fixes it.. I'm going to merge.  Thanks!